x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -338,7 +360,7 @@ func addHostFields(resultMetricsetFields common.MapStr, rootFields common.MapStr
 
 	for ec2MetricName, hostMetricName := range hostFieldTable {
 		metricValue, err := resultMetricsetFields.GetValue(ec2MetricName)
-		if ec2MetricName == "cpu.total.pct" {
+		if metricValue != nil && ec2MetricName == "cpu.total.pct" {
 			metricValue = metricValue.(float64) / 100
```

Comment:
This can be safer that only checking for `nil` value:
```suggestion
		if value, ok := metricValue.(float64); ok && ec2MetricName == "cpu.total.pct" {
			metricValue = value / 100
		}
```

Reply:
Thanks! I changed to 
```
                if value, ok := metricValue.(float64); ok {
			if ec2MetricName == "cpu.total.pct" {
				value = value / 100
			}
			rootFields.Put(hostMetricName, value)
		}
```